server.go

```diff
@@ -2024,138 +2027,153 @@ func (s *Server) ReadSeries(database, retentionPolicy, name string, tags map[str
+				res = s.executeShowServersStatement(stmt, user)
+			case *influxql.CreateUserStatement:
+				res = s.executeCreateUserStatement(stmt, user)
+			case *influxql.DeleteStatement:
+				res = s.executeDeleteStatement()
+			case *influxql.DropUserStatement:
+				res = s.executeDropUserStatement(stmt, user)
+			case *influxql.ShowUsersStatement:
+				res = s.executeShowUsersStatement(stmt, user)
+			case *influxql.DropSeriesStatement:
+				res = s.executeDropSeriesStatement(stmt, database, user)
+			case *influxql.ShowSeriesStatement:
+				res = s.executeShowSeriesStatement(stmt, database, user)
+			case *influxql.DropMeasurementStatement:
+				res = s.executeDropMeasurementStatement(stmt, database, user)
+			case *influxql.ShowMeasurementsStatement:
+				res = s.executeShowMeasurementsStatement(stmt, database, user)
+			case *influxql.ShowTagKeysStatement:
+				res = s.executeShowTagKeysStatement(stmt, database, user)
+			case *influxql.ShowTagValuesStatement:
+				res = s.executeShowTagValuesStatement(stmt, database, user)
+			case *influxql.ShowFieldKeysStatement:
+				res = s.executeShowFieldKeysStatement(stmt, database, user)
+			case *influxql.ShowStatsStatement:
+				res = s.executeShowStatsStatement(stmt, user)
+			case *influxql.GrantStatement:
+				res = s.executeGrantStatement(stmt, user)
+			case *influxql.RevokeStatement:
+				res = s.executeRevokeStatement(stmt, user)
+			case *influxql.CreateRetentionPolicyStatement:
+				res = s.executeCreateRetentionPolicyStatement(stmt, user)
+			case *influxql.AlterRetentionPolicyStatement:
+				res = s.executeAlterRetentionPolicyStatement(stmt, user)
+			case *influxql.DropRetentionPolicyStatement:
+				res = s.executeDropRetentionPolicyStatement(stmt, user)
+			case *influxql.ShowRetentionPoliciesStatement:
+				res = s.executeShowRetentionPoliciesStatement(stmt, user)
+			case *influxql.CreateContinuousQueryStatement:
+				res = s.executeCreateContinuousQueryStatement(stmt, user)
+			case *influxql.DropContinuousQueryStatement:
+				continue
+			case *influxql.ShowContinuousQueriesStatement:
+				res = s.executeShowContinuousQueriesStatement(stmt, database, user)
+			default:
+				panic(fmt.Sprintf("unsupported statement type: %T", stmt))
+			}
 
-		// If an error occurs then stop processing remaining statements.
-		results.Results[i] = res
-		if res.Err != nil {
-			break
+			if res != nil {
+				// set the StatementID for the handler on the other side to combine results
+				res.StatementID = i
+
+				// If an error occurs then stop processing remaining statements.
+				results <- res
+				if res.Err != nil {
+					break
+				}
+			}
 		}
-		s.stats.Inc("queriesExecuted")
-	}
 
-	// Fill any empty results after error.
-	for i, res := range results.Results {
-		if res == nil {
-			results.Results[i] = &Result{Err: ErrNotExecuted}
+		// if there was an error send results that the remaining statements weren't executed
+		for ; i < len(q.Statements)-1; i++ {
```

Comment:
Is this right? `i` is declared as a variable up above. It'll be incremented by the range statement. And yet it is then incremented here. What is going on?